django/test/runner.py

```diff
@@ -52,19 +56,247 @@ def printErrorList(self, flavour, errors):
             self.stream.writeln("%s" % sql_debug)
 
 
+class RemoteTestResult(object):
+    """
+    Record information about which tests have succeeded and which have failed.
+
+    The sole purpose of this class is to record events in the child processes
+    so they can be replayed in the master process. As a consequence it doesn't
+    inherit unittest.TestResult and doesn't attempt to implement all its API.
+
+    The implementation matches the unpythonic coding style of unittest2.
+    """
+
+    def __init__(self):
+        self.events = []
+        self.failfast = False
+        self.shouldStop = False
+        self.testsRun = 0
+
+    @property
+    def test_index(self):
+        return self.testsRun - 1
+
+    def check_pickleable(self, test, err):
+        # Ensure that sys.exc_info() tuples are picklable. This dislays a
+        # clear multiprocessing.pool.RemoteTraceback generated in the child
+        # process instead of a multiprocessing.pool.MaybeEncodingError, making
+        # the root cause easier to figure out for users who aren't familiar
+        # with the multiprocessing module. Since we're in a forked process,
+        # our best chance to communicate with them is to print to stdout.
+        try:
+            pickle.dumps(err)
+        except Exception:
+            print("""
+{} failed.
+
+Unfortunately, it raised an exception which cannot be pickled, making it
+impossible for the parallel test runner to display it properly.
+
+The error traceback below may help you identify the problem. Otherwise you can
+run the tests without the --parallel option.
+
+""".format(test))
```

Comment:
Ah, the traceback is already printed, but it's completely inside the test runner.

Indeed the only relevant information is the exception repr. I'm adding it into the text.